### PR TITLE
feat: add zipkin variable

### PR DESCRIPTION
**Pull request by wizhuo**
### Description

I would like to expose the zipkin span information during the rewrite phase so that it can be used in other plugins and scenarios. For example, it would be useful to include the trace ID when printing logs.

Fixes # (issue)

### Checklist

- [ ] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: apisix/plugins/zipkin.lua [hunk 1]
```diff
@@ -69,6 +78,11 @@ function _M.check_schema(conf)
     return core.schema.check(schema, conf)
 end
 
+local plugin_info  = plugin.plugin_attr(plugin_name) or {}
+
+
+
+
```
Comment (monkeyDluffy6017) [inline apisix/plugins/zipkin.lua:85]: Why so many blanks?

File: t/plugin/zipkin3.t [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+=== TEST 2:  trigger zipkin with open set variables
+--- request
+GET /echo
+--- error_log eval
+qr/ngx_var.zipkin_context_traceparent:00-\w{32}-\w{16}-01*/
+
+
+
+
+
```
Comment (monkeyDluffy6017) [inline t/plugin/zipkin3.t:123]: Three blanks are needed

File: apisix/plugins/zipkin.lua [hunk 3]
```diff
@@ -51,7 +55,12 @@ local schema = {
         span_version = {
             enum = {ZIPKIN_SPAN_VER_1, ZIPKIN_SPAN_VER_2},
             default = ZIPKIN_SPAN_VER_2,
-        },
+        },       
+        set_ngx_var = {
+            type = "boolean",
+            description = "set nginx variables",
+            default = false,
+          },
```
Comment (monkeyDluffy6017) [inline apisix/plugins/zipkin.lua:63]: Why do you add the check here? Are you going to pass this configuration in the plugin parameters?？It is unnecessary if you just want to define the configuration in `config.yaml`

File: t/cli/test_zipkin_set_ngx_var.sh [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    echo "failed: zipkin_trace_id not found in nginx.conf"
+    exit 1
+fi
+
+if ! grep "set \$zipkin_span_id                      '';" conf/nginx.conf > /dev/null; then
+    echo "failed: zipkin_span_id not found in nginx.conf"
+    exit 1
+fi
+
+
```
Comment (monkeyDluffy6017) [inline t/cli/test_zipkin_set_ngx_var.sh:45]: Please remove the redundant blank

**Comment by monkeyDluffy6017** [comment]
Good job! Please make the ci pass

**Comment by wizhuo** [comment]
> Good job! Please make the ci pass

I have already updated, please check

File: apisix/cli/ops.lua [hunk 5]
```diff
@@ -546,6 +546,11 @@ Please modify "admin_key" in conf/config.yaml .
     if enabled_plugins["opentelemetry"] and yaml_conf.plugin_attr["opentelemetry"] then
         opentelemetry_set_ngx_var = yaml_conf.plugin_attr["opentelemetry"].set_ngx_var
     end
+
+    local zipkin_set_ngx_var
+    if enabled_plugins["zipkin"] and yaml_conf.plugin_attr["zipkin"] then
+        zipkin_set_ngx_var = yaml_conf.plugin_attr["zipkin"].set_ngx_var
+    end
```
Comment (monkeyDluffy6017) [inline apisix/cli/ops.lua:553]: Need one blank

**Comment by monkeyDluffy6017** [comment]
@wizhuo please check the ci error

**Comment by wizhuo** [comment]
> @wizhuo please check the ci error

Please advise on how to resolve the following issue


Error: nginx: [error] open() "/home/runner/work/apisix/apisix/logs/nginx.pid" failed (2: No such file or directory)
[ info ] stop -> [ Done ]
+ git checkout conf/config.yaml
Updated 0 paths from the index
+ for f in ./t/cli/test_*.sh
+ PATH=/usr/local/openresty-debug/nginx/sbin:/usr/local/openresty-debug/luajit/bin:/usr/local/openresty-debug/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin
+ ./t/cli/test_zipkin_set_ngx_var.sh
./ci/linux_apisix_current_luarocks_runner.sh: line 74: ./t/cli/test_zipkin_set_ngx_var.sh: Permission denied
Error: Process completed with exit code 126.

**Comment by monkeyDluffy6017** [comment]
@wizhuo you can test in your private repository with the following configuration
```
  - name: Run debugger
    if: ${{ failure() }}
    uses: mxschmitt/action-tmate@v3
    timeout-minutes: 15
```

**Comment by monkeyDluffy6017** [comment]
@wizhuo Nice job!

**Review by monkeyDluffy6017 — APPROVED** [review]

**Review by AlinsRan — APPROVED** [review]

**Pull request merged**